nRFMeshProvision/Classes/ProxyFilter.swift

```diff
@@ -443,12 +451,21 @@ internal extension ProxyFilter {
                 return
             }
             counter = 0
+            callProxyFilterUpdateAcknowledged(type: status.filterType, addresses: addresses)
         default:
             // Ignore.
             break
         }
     }
-    
+
+    private func callProxyFilterUpdateAcknowledged(
+        type: ProxyFilerType,
+        addresses: Set<Address>
+    ) {
+        delegateQueue.async { [delegate] in
```

Comment:
It might seem overkill to create a function for these few lines, but what this function also does for us is capturing the current value of `addresses`.
Otherwise we would have to write something like this:

```swift
let delegateAddresses = addresses
dispatchQueue.async {
// .. same code 
}
```

Without the additional constant we couldn't be sure that `self.addresses` didn't change since we scheduled the call.

Reply:
That's OK, I understand. 
I also noticed, that you decided to pass `[delegate]` to the `async` block, instead of strong or weak `self`, like in other places. Do you think this is a better choice?

Reply:
I went with a strong reference so the delegate is called eventhough the filter is dead, so any process waiting for it does not timeout (and throw an error for the wrong reason).

And since we only need `delegate` and not all of `self`, I strong referenced `delegate` only.

What are your thoughts?